pandas/core/internals/blocks.py

```diff
@@ -684,7 +684,10 @@ def to_native_types(self, slicer=None, na_rep="nan", quoting=None, **kwargs):
         itemsize = writers.word_len(na_rep)
 
         if not self.is_object and not quoting and itemsize:
-            values = values.astype(f"<U{itemsize}")
+            values = values.astype(str)
+            if values.dtype.itemsize / np.dtype("U1").itemsize < itemsize:
```

Comment:
Any performance implications to this? An alternate strategy I think would be to compare to the `itemsize` to the length of the `na_rep` and take the max, which I think would reduce the amount of `astype` calls needed, assuming they come at a noticeable cost

Reply:
Can you reword your suggestion? I don't quite follow. In particular, I think the variable `itemsize` here *is* the length of the stringified na_rep.

My idea here was: we don't know what the string lengths will be without scanning the values and doing the actually conversion. So just let NumPy do that, and only astype to a larger dtype if needed.

Reply:
Hmm sorry I was mistakenly thinking that the issue was only truncating when all values in the array had a str length that was less than that of na_rep, which isn't the case

Do we even need the second astype?